### PR TITLE
Add Clinics system and controller testing

### LINKED ISSUE #1175: Write clinic controller and integration tests
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

What are we trying to do?
Clinic creation and maintenance is an area where our integration test coverage is pretty spotty. We should bump that.

What feature or behavior is this required for?
Having our acts together

How could we solve this issue? (Not knowing is okay!)
Would like the test to affirm the following:

Creating a clinic persists

Updating individual fields of a clinic all persist

We can see information about clinics

Anything else?
nope!

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

Atoning for not writing clinic tests when we rolled it out last year.

This pull request makes the following changes:
* Adds controller tests
* Adds a system test
* Makes a few DOM changes to ids to simplify the testing

No view changes.

It relates to the following issue #s: 
* Fixes #1175

### CODE REVIEW

File: test/controllers/clinics_controller_test.rb [hunk 1]
```diff
@@ -21,20 +22,22 @@ class ClinicsControllerTest < ActionDispatch::IntegrationTest
...
     before { @new_clinic = attributes_for :clinic }
 
-    # it 'should raise if user is not admin' do # TODO
-    #   @user.role = :cm
-    #   @user.save!
-    #   assert_raise RuntimeError, 'Permission Denied' do
-    #     post :create
-    #   end
-    # end
+    it 'should raise if user is not admin' do
```
Comment (lomky) [inline test/controllers/clinics_controller_test.rb]: I don't think this is raising anything. Wrong name?

File: test/system/clinic_management_test.rb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+  describe 'viewing clinic info' do
+    before { visit clinics_path }
+
+    it 'should have existing clinics in the table' do
+      within :css, '#clinics_table' do
+        assert has_text? @clinic.name
+      end
+
+      # TODO evisit this after it's less clunky
```
Comment (lomky) [inline test/system/clinic_management_test.rb]: Just noting the TODO. Did you mean to finish this for this PR? If not, maybe throw a followup PR in?

File: test/controllers/clinics_controller_test.rb (hunk 1 above)
Comment (colinxfleming) [reply test/controllers/clinics_controller_test.rb]: yep, good catch! retitling this test to reflect the redirection.

File: test/system/clinic_management_test.rb (hunk 2 above)
Comment (colinxfleming) [reply test/system/clinic_management_test.rb]: Good call! Created an issue for this in https://github.com/DCAFEngineering/dcaf_case_management/issues/1353 and added a bit to the note.

**Comment by colinxfleming** [comment]
thank you @lomky !

**Pull request merged**